### PR TITLE
Confirm-password

### PR DESCRIPTION
**Pull request by AlaaTaima**
Relates #8

### CODE REVIEW

File: src/containers/ConfirmPassword/confirmPass.jsx [hunk 1]
```diff
@@ -11,19 +9,22 @@ import { withFirebase } from '../Firebase';
...
+  const {
+    form: { getFieldDecorator },
+    handelGoBack,
+    handleChange,
+    handleErrorMessage,
+    handlePush,
+    errorMessage,
+  } = props;
+
+  const handleSubmit = e => {
```
Comment (yosefanajjar) [inline src/containers/ConfirmPassword/confirmPass.jsx]: Functions should be in the index folder

File: src/containers/ConfirmPassword/confirmPass.jsx [hunk 2]
```diff
@@ -33,85 +34,71 @@ class ConfirmPass extends Component {
             values.password
           );
           localStorage.setItem('confirm', true);
-          await push(EDIT_ACCOUNT);
+          await handlePush(EDIT_ACCOUNT);
```
Comment (yosefanajjar) [inline src/containers/ConfirmPassword/confirmPass.jsx]: We don't need **await** here

File: src/containers/ConfirmPassword/confirmPass.jsx [hunk 3]
```diff
@@ -33,85 +34,71 @@ class ConfirmPass extends Component {
...
         } catch (error) {
-          this.setState({ errorMesage: 'Invalid password' });
+          if (error.message) handleErrorMessage(error.message);
```
Comment (yosefanajjar) [inline src/containers/ConfirmPassword/confirmPass.jsx]: I think you should check for the **error** not the message:
```suggestion
          if (error) handleErrorMessage(error.message);
```

File: src/containers/ConfirmPassword/confirmPass.jsx [hunk 4]
```diff
@@ -33,85 +34,71 @@ class ConfirmPass extends Component {
...
   }).isRequired,
 };
 
-const AuthConfirmaPassword = compose(
-  withAuth,
-  withFirebase
-)(confirmPasswordForm);
-
-export default AuthConfirmaPassword;
+export default withFirebase(confirmPasswordForm);
```
Comment (yosefanajjar) [inline src/containers/ConfirmPassword/confirmPass.jsx]: You are using firebase in the index file which is enough so if you want anything fro firebase to use here you can **pass it via the props** but I think if you move that function you will not even need withFirebase here

File: src/containers/ConfirmPassword/index.js [hunk 5]
```diff
@@ -1,3 +1,50 @@
...
+import { withFirebase } from '../Firebase';
 import ConfirmPassword from './confirmPass';
 
-export default ConfirmPassword;
+class Index extends Component {
+  state = {
+    errorMessage: '',
+  };
+
+  handleChange = () => {
```
Comment (yosefanajjar) [inline src/containers/ConfirmPassword/index.js:45]: Why do we need this function?

Comment (Fatmasiam) [reply src/containers/ConfirmPassword/index.js:45]: to empty the error message when the user writes on the input 
--> the error msg comes from the firebase 
@yosefalnajjarofficial

File: src/containers/ConfirmPassword/index.js [hunk 6]
```diff
@@ -1,3 +1,94 @@
...
+          const user = await firebase.auth.currentUser;
+          await firebase.doSignInWithEmailAndPassword(
+            user.email,
+            values.password
+          );
+          localStorage.setItem('confirm', true);
+          this.setState({ loading: false });
+          push(EDIT_ACCOUNT);
+        } catch (error) {
+          if (error) this.setState({ errorMessage: error.message });
```
Comment (Fatmasiam) [inline src/containers/ConfirmPassword/index.js]: here also you should setState to the loading to be false
```suggestion
          if (error) this.setState({ errorMessage: error.message, loading: false  });
```

File: src/containers/ConfirmPassword/confirmPass.jsx [hunk 7]
```diff
@@ -1,117 +1,79 @@
...
+
+          {errorMessage && <p style={{ color: 'red' }}>{errorMessage}</p>}
+
+          <Button
+            type="primary"
+            size="large"
+            className="confirm-pass__form__loading"
+            onClick={handleSubmit}
+          >
+            {loading ? <Spin /> : 'Proceed To Edit'}
```
Comment (Fatmasiam) [inline src/containers/ConfirmPassword/confirmPass.jsx:57]: put a class to the <Spin /> to make the color to it white no orange

**Review by yosefanajjar — APPROVED** [review]

**Review by Fatmasiam — APPROVED** [review]

**Pull request merged**